src/nl/hannahsten/texifyidea/run/pdfviewer/PdfViewer.kt

```diff
@@ -12,49 +14,102 @@ import com.intellij.openapi.project.Project
 private val EP_NAME = ExtensionPointName<ExternalPdfViewer>("nl.rubensten.texifyidea.pdfViewer")
 
 /**
- * Interface that defines a pdf viewer so we can use both [nl.hannahsten.texifyidea.run.linuxpdfviewer.InternalPdfViewer]s and [ExternalPdfViewer]s interchangeably.
+ * Interface that defines the general behavior of a PDF viewer.
  */
 interface PdfViewer {
 
+    /**
+     * The identifier of the viewer.
+     */
     val name: String?
     val displayName: String?
 
+    /**
+     * Tells whether the PDF viewer is available.
+     */
     fun isAvailable(): Boolean
 
     /**
-     * Performs a forward search to navigate from the source file to the corresponding location in the PDF document.
+     * Whether the PDF viewer supports forward search.
+     */
+    val isForwardSearchSupported: Boolean
+        get() = false
```

Comment:
I will put the default to true for now to not break the pdf viewer plugin compatibility. I can add an explicit override there, with the disadvantage that we cannot easily change this property anymore